help_to_heat/frontdoor/views.py

```diff
@@ -419,37 +537,70 @@ def get_context(self, request, session_id, *args, **kwargs):
 class EpcView(PageView):
     def get_context(self, request, session_id, page_name, data):
         session_data = interface.api.session.get_session(session_id)
-        uprn = session_data.get("uprn")
-        address = session_data.get("address")
         country = session_data.get("country")
-        if uprn:
-            epc = interface.api.epc.get_epc(uprn, country)
+
+        if country == "Scotland":
+            uprn = session_data.get("uprn")
+            address = session_data.get("address")
+            country = session_data.get("country")
+            if uprn:
+                epc = interface.api.epc.get_epc_scotland(uprn, country)
+            else:
+                epc = {}
+            context = {
+                "epc_rating": epc.get("rating"),
+                "epc_date": epc.get("date"),
+                "epc_display_options": schemas.epc_display_options_map,
+                "address": address,
+            }
+            return context
         else:
-            epc = {}
-        context = {
-            "epc_rating": epc.get("rating"),
-            "epc_date": epc.get("date"),
-            "epc_display_options": schemas.epc_display_options_map,
-            "address": address,
-        }
-        return context
+            rrn = session_data.get("rrn")
+            address = session_data.get("address")
+            context = {}
+            if rrn:
+                epc = session_data.get("epc_details")
+            else:
+                epc = {}
+
+            context = {
+                "epc_rating": epc.get("currentEnergyEfficiencyBand").upper()
+                if epc.get("currentEnergyEfficiencyBand")
+                else "",
+                "epc_date": epc.get("lodgementDate"),
+                "epc_display_options": schemas.epc_display_options_map,
+                "address": address,
+            }
+            return context
 
     def handle_get(self, response, request, session_id, page_name, context):
         session_data = interface.api.session.get_session(session_id)
-        uprn = session_data.get("uprn")
         country = session_data.get("country")
-        uprn = session_data.get("uprn")
-        if uprn:
-            epc = interface.api.epc.get_epc(uprn, country)
+
+        if country == "Scotland":
+            uprn = session_data.get("uprn")
+            country = session_data.get("country")
+            uprn = session_data.get("uprn")
+            if uprn:
+                epc = interface.api.epc.get_epc_scotland(uprn, country)
+            else:
+                epc = {}
+            if not epc:
+                return redirect("frontdoor:page", session_id=session_id, page_name="benefits")
+            return super().handle_get(response, request, session_id, page_name, context)
```

Comment:
Re-reading this, if what I have above is correct, I think it can be further simplified to:
```
reference_number = session_data.get("uprn") if country == "Scotland" else session_data.get("rrn")
if reference_number:
    return super().handle_get(response, request, session_id, page_name, context)
else:
    return redirect("frontdoor:page", session_id=session_id, page_name="benefits")
```